### PR TITLE
[ISSUE #5738] remove uselesscode in updateFaultItem

### LINKED ISSUE #5738: [code optimization] simplify the method updateFaultItem
https://github.com/apache/rocketmq/blob/e77d0673f8f96c7aff3a2252d79f87952539c7e4/client/src/main/java/org/apache/rocketmq/client/latency/LatencyFaultToleranceImpl.java#L30-L49

the logic is to update the existed one or ,put the new item.

and can be simpilified into 
pseudocode:
```
var item  = new FaultIem();
init(item)
put(name,item)

```

### PR DESCRIPTION
**Pull request by xiaoyifang**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #5738

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by mxsm — APPROVED** [review]
LGTM~

File: client/src/main/java/org/apache/rocketmq/client/latency/LatencyFaultToleranceImpl.java [hunk 1]
```diff
@@ -37,11 +37,7 @@ public void updateFaultItem(final String name, final long currentLatency, final
             faultItem.setCurrentLatency(currentLatency);
             faultItem.setStartTimestamp(System.currentTimeMillis() + notAvailableDuration);
 
-            old = this.faultItemTable.putIfAbsent(name, faultItem);
-            if (old != null) {
-                old.setCurrentLatency(currentLatency);
-                old.setStartTimestamp(System.currentTimeMillis() + notAvailableDuration);
-            }
+            this.faultItemTable.putIfAbsent(name, faultItem);
```
Comment (zhouxinyu) [inline client/src/main/java/org/apache/rocketmq/client/latency/LatencyFaultToleranceImpl.java:40]: The original code is used to update the latency to exists fault item. Could you please write a test case to verify your issue?

**Review by zhouxinyu — CHANGES_REQUESTED** [review]

File: client/src/main/java/org/apache/rocketmq/client/latency/LatencyFaultToleranceImpl.java (hunk 1 above)
Comment (xiaoyifang) [reply client/src/main/java/org/apache/rocketmq/client/latency/LatencyFaultToleranceImpl.java:40]: you are right.

**Comment by xiaoyifang** [comment]
the original code is ok.

**Comment by xiaoyifang** [comment]
rewrite the method

**Pull request closed without merging**